### ISSUE TITLE
Proposal of new Zanimo API

### ISSUE THREAD

**Issue by gre** [peutetre/Zanimo#10]
I've played a lot with this awesome library when making https://github.com/gre/ld28 .

and,
Here is a proposal of API refactoring.

**The idea here is to export Zanimo as one function: the transition function which is the core of what Zanimo solves.**

Here is today's Zanimo API:

``` javascript
[1] Zanimo (domElt) => Promise[DOM]
[2] Zanimo.transition (domElt, attr, value, duration, timing) => Promise[DOM]
[3] Zanimo.transitionf (attr, value, duration, timing) => (Function[DOM] => Promise[DOM])
[4] Zanimo.transform (elt, value, overwrite) => Promise[DOM]
[5] Zanimo.transformf (value, overwrite) =>  (Function[DOM] => Promise[DOM])
[6] Zanimo.f (elt) =>  (Function[DOM] => Promise[DOM])
[7] Zanimo.all (flist) => Promise[_]
```

**tl;dr. My proposal is to export only the `Zanimo.transition` function into `Zanimo` and also to preserve `transitionf` as an utility function into `Zanimo.f`.**

We could group features of [1], [2] and [4] in one function:

``` javascript
var Zanimo = require("zanimo-proposal");
Zanimo(elt, "transform", "100px");
Zanimo(elt) // if taking one argument, just doing a requestAnimFrame
  .then(function (elt) {
    return Zanimo(elt, "transform", "100px", "1s"/*, "linear"*/);
  })
```

So there would be 4 signatures possible which always return the Promise of the element:

``` javascript
[a] Zanimo(elt)
[b] Zanimo(elt, attr, value)
[c] Zanimo(elt, attr, value, duration)
[d] Zanimo(elt, attr, value, duration, easing)
```

The `[b]` signature of this proposal does not have any duration, it means you want to do it now, which is very like in [4] but **for any property**. However it doesn't have the overwrite feature anymore (is that feature required?).

Also I don't forget the useful `transitionf` feature and I think we can call it `f`,
which is still just a lazy call to this proposal new Zanimo function:

``` javascript
Zanimo.f = function (attr, value, duration, timing) {
  var args = Array.prototype.slice.call(arguments);
  return function (elt) {
      return Zanimo.apply(this, [elt].concat(args));
  };
};
```

Concerning the function [6], we can substitute it by this Q code:

``` javascript
// .then(Zanimo.f(elt))
.thenResolve(elt)
```

Concerning [7], maybe we can keep it, but the use-case is for me a bit specific,
I would suggest to let the final user use Q.all.

e.g. https://github.com/gre/ld28/blob/master/src/games/calculation.js#L455-L459

What do you think? I can work on a PR when it's ok for you ;)

Thanks,
gre
